### ISSUE TITLE
Shibuya archive block 330827 returning errors

### ISSUE THREAD

**Issue by XY-Wang** [subsquid/archive-gateway#3]
Hi guys,

Shibuya Fire Squid archive is returning this error at block 330827

```json
{
  "errors": {
    "message": "JSON.parse: unexpected end of data at line 1 column 1 of the JSON data",
    "stack": ""
  }
}
```

This is the query I'm executing in the [aquarium](https://app.subsquid.io/aquarium/archives/shibuya/subsquid)
```json5
query {
  status {
    head
  }
  batch(limit: 1, fromBlock: 330827, events: {name: "Balances.Transfer"}) {
    calls
    events
    extrinsics
  }
}
```

From https://polkadot.js.org/apps/#/explorer/query/330827 it seems that there is a `utility.batch` extrinsic with lots of `balance.transfer` calls. Maybe it is related?

**Comment by tmcgroul** [comment]
Hi
Thank you for the report!

I ran the gateway on my computer then executed the query and found that it consumed more memory than we dedicated to our hosted gateway.
Will try to optimize the gateway or dedicate more memory

**Comment by XY-Wang** [comment]
Thanks for checking it out and the explanation! Looking forward to the fix :wink:

**Comment by tmcgroul** [comment]
I made a fix which is available in subsquid/substrate-gateway:1.2.1 and https://shibuya.archive.subsquid.io/graphql as well
The gateway used to send inefficient queries to a database and it somehow consumed a lot of memory. Now should be fine

**Comment by XY-Wang** [comment]
Thank you, it works like a charm! Closing the issue.
